daemon/graphdriver/overlay/overlay.go

```diff
@@ -263,6 +300,20 @@ func (d *Driver) Create(id, parent string, opts *graphdriver.CreateOpts) (retErr
 		}
 	}()
 
+	driver := &Driver{}
+	if opts != nil {
+		if err := d.parseStorageOpt(opts.StorageOpt, driver); err != nil {
+			return err
+		}
+	}
+
+	if driver.options.quota.Size > 0 && !strings.HasSuffix(dir, "-init") {
+		// enforce container disk project quota
+		if err := d.quotaCtl.SetQuota(dir, driver.options.quota); err != nil {
```

Comment:
@albamc The issue I think you are having on setting this on the upper dir is you are setting it on *all* upper dirs. This causes errors about cross-device linking.

If you change this to only apply when building the rw layer, setting only the upper dir seems to work well.